File: results/dnd-rest-benchmark/lifecycle-runs/20260731T160058Z_claude_claude-sonnet-5_typescript-vite/src/domain/storage.ts
/**
 * Storage lifecycle: opens the SQLite database and rehydrates the in-memory
 * caches owned by the auth and combat modules, and exposes the status/reset
 * endpoints used by tests to inspect or wipe persisted state.
 */

import { getDb, isInitialized, resetStorage, SCHEMA_VERSION } from '../db.ts';
import type { ApiResult } from '../types.ts';
import { clearUsers, loadUsersFromDb } from './auth.ts';
import { clearCombatSessions, loadCombatSessionsFromDb } from './combat.ts';

export function initStorage(): void {
  getDb();
  loadUsersFromDb();
  loadCombatSessionsFromDb();
}

export function storageStatus(): ApiResult {
  return {
    status: 200,
    body: { driver: 'sqlite', schema_version: SCHEMA_VERSION, initialized: isInitialized() },
  };
}

export function resetStorageHandler(): ApiResult {
  resetStorage();
  clearUsers();
  clearCombatSessions();
  return { status: 200, body: { ok: true, schema_version: SCHEMA_VERSION } };
}

File: results/dnd-rest-benchmark/lifecycle-runs/20260730T075550Z_pi_kimi-k2p7-code_ruby-sinatra/lib/routes/storage.rb
# frozen_string_literal: true

# Route definitions for storage endpoints.

# --- Storage ---

get '/v1/storage/status' do
  content_type :json
  JSON.dump(
    driver: 'sqlite',
    schema_version: Storage::SCHEMA_VERSION,
    initialized: Storage.initialized?
  )
end

post '/v1/storage/reset' do
  content_type :json
  begin
    Storage.reset!
    JSON.dump(ok: true, schema_version: Storage::SCHEMA_VERSION)
  rescue SQLite3::Exception
    json_error(500, 'storage reset failed')
  end
end

